homeassistant/components/bmw_connected_drive/number.py

```diff
@@ -99,14 +99,14 @@ def __init__(
         self._attr_native_value = description.value_fn(vehicle)
         self._attr_mode = description.mode
 
-    @callback
-    def _handle_coordinator_update(self) -> None:
-        """Handle updated data from the coordinator."""
-        _LOGGER.debug(
-            "Updating Number '%s' of %s", self.entity_description.key, self.vehicle.name
-        )
-        self._attr_native_value = self.entity_description.value_fn(self.vehicle)
-        super()._handle_coordinator_update()
+    @property
+    def native_value(self) -> float | None:
+        """Return the entity value to represent the entity state."""
+        # Raw value
+        if (value := self.entity_description.value_fn(self.vehicle)) is None:
```

Comment:
We don't seem to need the check. We can just return the value regardless if it's None or not.